src/components/layout/Button.tsx

```diff
@@ -1,69 +1,62 @@
-                glowy: "bg-black/10 shadow-xl text-white border border-m-blue hover:m-blue-dark hover:text-m-blue",
-                blue: "bg-m-blue text-white shadow-inner-button hover:bg-m-blue-dark text-shadow-button",
-                red: "bg-m-red text-white shadow-inner-button hover:bg-m-red-dark text-shadow-button",
-                green: "bg-m-green text-white shadow-inner-button hover:bg-m-green-dark text-shadow-button",
-                text: ""
-            },
-            layout: {
-                flex: "flex-1 text-xl",
-                pad: "px-8 text-xl",
-                small: "px-4 py-2 w-auto text-lg",
-                xs: "px-4 py-2 w-auto rounded-lg text-base",
-                full: "w-full text-xl"
-            }
-        },
-        defaultVariants: {
-            intent: "inactive",
-            layout: "flex"
-        }
-    }
-);
-
 // Help from https://github.com/arpadgabor/credee/blob/main/packages/www/src/components/ui/button.tsx
-
-type StyleProps = VariantProps<typeof button>;
-interface ButtonProps
-    extends JSX.ButtonHTMLAttributes<HTMLButtonElement>,
-        StyleProps {
+export interface CommonButtonStyleProps {
     loading?: boolean;
     disabled?: boolean;
+    intent?:
+        | "active"
+        | "inactive"
+        | "glowy"
+        | "blue"
+        | "red"
+        | "green"
+        | "text";
+    layout?: "flex" | "pad" | "small" | "xs" | "full";
 }
 
-export const Button: ParentComponent<ButtonProps> = (props) => {
-    const slot = children(() => props.children);
-    const [local, attrs] = splitProps(props, [
-        "children",
-        "intent",
-        "layout",
-        "class"
-    ]);
+export type ButtonProps = CommonButtonStyleProps & {
+    onClick?: (e: MouseEvent | SubmitEvent) => void;
+    type?: string;
+};
+
+export type ButtonLinkProps = CommonButtonStyleProps & {
+    href: string;
+    target?: string;
+    rel?: string;
+};
 
+export const Button: ParentComponent<ButtonProps> = (props) => {
     return (
         <button
-            {...attrs}
+            onClick={props.onClick}
             disabled={props.disabled || props.loading}
-            class={button({
-                class: local.class || "",
-                intent: local.intent,
-                layout: local.layout
-            })}
+            class="p-3 rounded-xl font-semibold transition disabled:bg-neutral-400/10 disabled:text-white/20 disabled:shadow-inner-button-disabled"
+            classList={{
+                "bg-white text-black border border-white hover:text-[#3B6CCC]":
+                    props.intent === "active",
+                "bg-transparent text-white border border-white hover:text-[#3B6CCC]":
+                    !props.intent || props.intent === "inactive",
+                "bg-black/10 shadow-xl text-white border border-m-blue hover:m-blue-dark hover:text-m-blue":
+                    props.intent === "glowy",
```

Comment:
we can get rid of "glowy" for this and button link, it's only used in KitchenSink fine to delete from its uses there and just have default button

Reply:
want to try intent="red" for those?